test/drbgtest.c

```diff
@@ -799,18 +799,15 @@ static void run_multi_thread_test(void)
 {
     unsigned char buf[256];
     time_t start = time(NULL);
-    RAND_DRBG *public, *private;
+    RAND_DRBG *public = NULL, *private = NULL;
```

Comment:
nit: is this initialization really necessary?

Reply:
Technically it would not be necessary, but since the control-flow with
TEST_ptr below could theoretically skip the initialization of private,
it triggers often a warning about uninitailized warnings, which makes
--strict-warnings build fail.
And in fact, the compiler cannot prove that TEST_ptr(p) != 0 if p != NULL.
We know that, but only because we known what the TEST_xxx functions do.
I proposed a while ago a different TEST_macro that would allow
the compiler to better predict the control flow, but that (PR #3322) was rejected.
Therefore it is necessary to initialize all values that are assigned in
if (TEST_... statements.

Reply:
ok, thanks for the explanation.